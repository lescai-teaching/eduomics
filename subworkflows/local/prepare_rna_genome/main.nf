include { SUBSETFASTATX      } from '../../../modules/local/subsetfastatx/main'
include { SUBSETGFF          } from '../../../modules/local/subsetgff/main'
include { SALMON_INDEX       } from '../../../modules/nf-core/salmon/index/main'

workflow PREPARE_RNA_GENOME {

    take:

    ch_fasta   // channel: [ path(fasta) ]
    ch_txfasta // channel: [ path(txfasta) ]
    ch_gff3    // channel: [ path(gff3) ]
    ch_meta    // channel: [ val(meta) ]

    main:

    ch_versions = Channel.empty()

    SUBSETFASTATX ( ch_meta, ch_txfasta, ch_gff3 )
    ch_versions = ch_versions.mix(SUBSETFASTATX.out.versions.first())

    SUBSETGFF ( ch_meta, ch_gff3 )
    ch_versions = ch_versions.mix(SUBSETGFF.out.versions.first())

    SALMON_INDEX ( ch_fasta, SUBSETFASTATX.out.fasta )
    ch_versions = ch_versions.mix(SALMON_INDEX.out.versions.first())

    emit:
    txfasta_index      = SALMON_INDEX.out.index          // channel: [ path(txfasta_index) ]
    txfasta            = SUBSETFASTATX.out.fasta         // channel: [ path(txfasta) ]
    gene_lists         = SUBSETGFF.out.geneLists         // channel: [ path(geneLists) ]
    transcript_data    = SUBSETGFF.out.transcriptData    // channel: [ path(transcriptData) ]

    versions = ch_versions                               // channel: [ versions.yml ]
}

