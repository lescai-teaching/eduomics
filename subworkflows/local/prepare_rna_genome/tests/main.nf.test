nextflow_workflow {

    name "Test Subworkflow PREPARE_RNA_GENOME"
    script "../main.nf"
    workflow "PREPARE_RNA_GENOME"

    config "./test.config"

    tag "subworkflows"
    tag "subworkflows_"
    tag "subworkflows/prepare_rna_genome"
    tag "subsetfastatx"
    tag "subsetgff"
    tag "salmon"
    tag "salmon/index"


    test("prepare_rna_genome - test") {

        when {

            workflow {
                """

                input[0] = file("${params.igenomes_base}/Homo_sapiens/NCBI/GRCh38/Sequence/WholeGenomeFasta/genome.fa")
                input[1] = file('https://github.com/lescai-teaching/eduomics_testdata/raw/refs/heads/main/rnaseq/subsetfastatx/gencode_transcripts_noversion_chr21_chr22.fasta', checkIfExists: true)
                input[2] = file('https://raw.githubusercontent.com/lescai-teaching/eduomics_testdata/refs/heads/main/rnaseq/subsetfastatx/gencode_chr21_chr22_v47.gff3', checkIfExists: true)
                input[3] = [ id:'test', chromosome:'chr22' ]

                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
