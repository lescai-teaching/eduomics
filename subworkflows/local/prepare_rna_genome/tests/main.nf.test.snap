{
    "prepare_rna_genome - test": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    "gencode_transcripts_noversion_chr22.fasta:md5,4b4d69b3e46a9c3d47cd94d55ce13d21"
                ],
                "2": [
                    "valid_gene_lists.rds:md5,d7c8a8267989395676bf144e7a793369"
                ],
                "3": [
                    "transcript_data.rds:md5,2829de5c7c53afd3a671c0e131101f6b"
                ],
                "4": [
                    "versions.yml:md5,2654c7a6b69615ffaf26e89ad773fca2",
                    "versions.yml:md5,cee755a176b8939ac905cca4ca3766c1"
                ],
                "gene_lists": [
                    "valid_gene_lists.rds:md5,d7c8a8267989395676bf144e7a793369"
                ],
                "transcript_data": [
                    "transcript_data.rds:md5,2829de5c7c53afd3a671c0e131101f6b"
                ],
                "txfasta": [
                    "gencode_transcripts_noversion_chr22.fasta:md5,4b4d69b3e46a9c3d47cd94d55ce13d21"
                ],
                "txfasta_index": [
                    
                ],
                "versions": [
                    "versions.yml:md5,2654c7a6b69615ffaf26e89ad773fca2",
                    "versions.yml:md5,cee755a176b8939ac905cca4ca3766c1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-03-26T14:42:14.906568963"
    }
}